public/app/percona/backup/components/ChunkedLogsViewer/ChunkedLogsViewer.test.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { asyncAct } from 'app/percona/shared/helpers/testUtils';
+import { render, screen } from '@testing-library/react';
+import { BackupLogs } from '../../Backup.types';
+import { ChunkedLogsViewer } from './ChunkedLogsViewer';
+import { Messages } from './ChunkedLogsViewer.messages';
+import userEvent from '@testing-library/user-event';
+
+describe('ChunkedLogsViewer', () => {
+  const getMockedLogsGetter = (logs: BackupLogs, timeout = 10): jest.Mock => {
+    return jest.fn().mockReturnValue(new Promise((resolve) => setTimeout(() => resolve(logs), timeout)));
+  };
+  beforeEach(() => {
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+
+  it('should show processing state in the beginning', () => {
+    const getLogs = getMockedLogsGetter({ logs: [], end: true });
+    render(<ChunkedLogsViewer getLogChunks={getLogs} />);
+    expect(screen.getByText(Messages.pleaseRefresh)).toBeInTheDocument();
+  });
+
+  it('should show "no logs" message after loading is done', async () => {
+    const getLogs = getMockedLogsGetter({ logs: [], end: true });
+    await asyncAct(() => {
```

Comment:
Do we still need `asyncAct`? Check `waitFor` from testing-library to see if we can remoce `asyncAct` and `advanceTimersByTime`. See a usage example [here](https://github.com/percona-platform/grafana/blob/14b1bf195b49af3804e6f4c7a2367a4d90dcdbe6/public/app/percona/dbaas/components/DBCluster/UpdateDBClusterModal/UpdateDBClusterModal.test.tsx).

Reply:
Good one, didn't know about it, thanks!